### ISSUE TITLE
Differentiate download options on complex projects

### ISSUE THREAD

**Issue by brightbold** [North101/mpc_projects#38]
In beta testing last Nov/Dec, @StormyWaters2021 suggested more and better differentiated download checkboxes.

StormyWaters:
> I don't know what the behind-the-scenes looks like, but it would be cool if everything had a more granular breakdown of what to print. If it's something I would know how to do, I can help with that part as well.

Coldtoes:
> It’s easy to break project down into more parts, it’s just a balance of what keeps the selections from being overwhelming. My theory has been that since it’s easy to delete individual cards once you’ve loaded a project in the extension, that that’s better than having a mile-long list of checkboxes. But the people may disagree!

StormyWaters:
> Fair enough! But as it currently stands, you have to add both Parallel and Mix and Match both, then go through and delete a bunch of stuff when I think a single extra checkbox [separating out signatures] for each gator would eliminate all that.

> So here is what I would do roughly if I were organizing the Parallel Mix and Match:
<img src="https://github.com/North101/mpc_projects/assets/405068/9a277b0b-4d2f-4fa9-8252-005ac74dc43e" width="550" />

> It does add length to the list, but I think indenting the sub-options keeps it clean and easy to understand. I think it would probably be doable to just merge the two Parallel projects into one though

```Parallel Investigators
-------------------------
[✔] Core Set (All 5) (5 Cards)
    [✔] + Signature Cards (+X Cards)
    [✔] + Mix and Match (+X Cards)
        [✔] Use Revised Core Art (+0 Cards)
        [✔] Swap Wendy Adams w/ Hard Knocks Alternate Art (+0 Cards)
    [ ] + Challenge Scenarios (+X Cards)

[✔] Dunwich Legacy (Jim Culver, Ashcan Pete, Zoey Samaras) (3 Cards)
    [ ] + Signature Cards (Jim and Zoey) (+X Cards)
    [ ] + Mix and Match (Jim and Zoey) (+X Cards)
    [✔] + Challenge Scenarios (Jim) (+X Cards)
```
> This is definitely not as compact of a list, but I think it's clear enough to convey the options and provides a good amount of customization for players.

> I like the breakdown on Taboos, but if it were me I would just split the current and old for a more obvious delineation:

```Current Taboo (v2.1 Aug 2023)
------------------------------
[ ] Complete Pack (119)
[ ] Changes from 2.0 Only (39)
[✔] Mutations Only (95)
[ ] New/Changed Mutations Only (28)

Previous Taboo (v2.0 Aug 2022)
------------------------------
[ ] Complete Pack (95)
[ ] Changes from 1.8 Only (40)
```
Coldtoes:
> To address one small piece: separating out Mix & Match was based on the assumption that most people don't print these. (I had feedback from people who went to print parallels and were surprised when they ended up with multiple copies of each investigator and they were kind of startled!)

> If that assumption is correct, I figured it would be better to inconvenience the small group of expert players, making them download two different projects, than confuse a larger group of players, some of whom are less experienced and don't really know what choices to make.

> So, to test whether I'm making a bad assumption, today I posted this poll inspired by you: ⁠
<img src="https://github.com/North101/mpc_projects/assets/405068/e1511c7e-3e9a-4e57-a9e0-4f143b9683e0" width="700" />

> For the other changes to the checkbox list — these are good suggestions and would probably clarify choices for people. ...We'd just need to figure out whether the increase in complexity is worth it to benefit the small number of complex projects (I think right now mainly Parallel and Taboo are likely to benefit from this, yeah?)

Recently, I brought this up again with @North101. Taboo and Parallel are getting pretty unwieldy, so I think it's time to think about this more seriously. Personally, I find indented checkboxes kind of cluttered, so I'm included to so more with headers and vertical separations. I see a couple options for how to do this:

1. Add yet another grouping, below parts (maybe group?) to the json structure. And/or: 
2. Just add some properties that serve no purpose other than styling, and then just insert them manually to add classes that can be used to add space before a paragraph or to indent text or whatever.

**Comment by brightbold** [comment]
In beta testing last Nov/Dec, @StormyWaters2021 suggested more and better differentiated download checkboxes.

StormyWaters:
> I don't know what the behind-the-scenes looks like, but it would be cool if everything had a more granular breakdown of what to print. If it's something I would know how to do, I can help with that part as well.

Coldtoes:
> It’s easy to break project down into more parts, it’s just a balance of what keeps the selections from being overwhelming. My theory has been that since it’s easy to delete individual cards once you’ve loaded a project in the extension, that that’s better than having a mile-long list of checkboxes. But the people may disagree!

StormyWaters:
> Fair enough! But as it currently stands, you have to add both Parallel and Mix and Match both, then go through and delete a bunch of stuff when I think a single extra checkbox [separating out signatures] for each gator would eliminate all that.

> So here is what I would do roughly if I were organizing the Parallel Mix and Match:
<img src="https://github.com/North101/mpc_projects/assets/405068/9a277b0b-4d2f-4fa9-8252-005ac74dc43e" width="550" />

> It does add length to the list, but I think indenting the sub-options keeps it clean and easy to understand. I think it would probably be doable to just merge the two Parallel projects into one though

```Parallel Investigators
-------------------------
[✔] Core Set (All 5) (5 Cards)
    [✔] + Signature Cards (+X Cards)
    [✔] + Mix and Match (+X Cards)
        [✔] Use Revised Core Art (+0 Cards)
        [✔] Swap Wendy Adams w/ Hard Knocks Alternate Art (+0 Cards)
    [ ] + Challenge Scenarios (+X Cards)

[✔] Dunwich Legacy (Jim Culver, Ashcan Pete, Zoey Samaras) (3 Cards)
    [ ] + Signature Cards (Jim and Zoey) (+X Cards)
    [ ] + Mix and Match (Jim and Zoey) (+X Cards)
    [✔] + Challenge Scenarios (Jim) (+X Cards)
```
> This is definitely not as compact of a list, but I think it's clear enough to convey the options and provides a good amount of customization for players.

> I like the breakdown on Taboos, but if it were me I would just split the current and old for a more obvious delineation:

```Current Taboo (v2.1 Aug 2023)
------------------------------
[ ] Complete Pack (119)
[ ] Changes from 2.0 Only (39)
[✔] Mutations Only (95)
[ ] New/Changed Mutations Only (28)

Previous Taboo (v2.0 Aug 2022)
------------------------------
[ ] Complete Pack (95)
[ ] Changes from 1.8 Only (40)
```
Coldtoes:
> To address one small piece: separating out Mix & Match was based on the assumption that most people don't print these. (I had feedback from people who went to print parallels and were surprised when they ended up with multiple copies of each investigator and they were kind of startled!)

> If that assumption is correct, I figured it would be better to inconvenience the small group of expert players, making them download two different projects, than confuse a larger group of players, some of whom are less experienced and don't really know what choices to make.

> So, to test whether I'm making a bad assumption, today I posted this poll inspired by you: ⁠
<img src="https://github.com/North101/mpc_projects/assets/405068/e1511c7e-3e9a-4e57-a9e0-4f143b9683e0" width="700" />

> For the other changes to the checkbox list — these are good suggestions and would probably clarify choices for people. ...We'd just need to figure out whether the increase in complexity is worth it to benefit the small number of complex projects (I think right now mainly Parallel and Taboo are likely to benefit from this, yeah?)

Recently, I brought this up again with @North101. Taboo and Parallel are getting pretty unwieldy, so I think it's time to think about this more seriously. Personally, I find indented checkboxes kind of cluttered, so I'm included to so more with headers and vertical separations. I see a couple options for how to do this:

1. Add yet another grouping, below parts (maybe group?) to the json structure. And/or: 
2. Just add some properties that serve no purpose other than styling, and then just insert them manually to add classes that can be used to add space before a paragraph or to indent text or whatever.

**Comment by North101** [comment]
Fix in 610f79af8d7c7c9e89264af77c81dd33dde1569d
